src/command/manifest/sync.ts

```diff
@@ -0,0 +1,68 @@
+import { readFileSync } from 'fs'
+import { Argument, LeafCommand, Option } from 'furious-commander'
+import { Reference } from 'mantaray-js'
+import { pickStamp } from '../../service/stamp'
+import { readdirDeepAsync } from '../../utils'
+import { stampProperties } from '../../utils/option'
+import { ManifestCommand } from './manifest-command'
+
+export class Sync extends ManifestCommand implements LeafCommand {
+  public readonly name = 'sync'
+  public readonly description = 'Sync a local folder to an existing manifest'
+
+  @Argument({ key: 'address', description: 'Root manifest reference', required: true })
+  public reference!: string
+
+  @Argument({ key: 'folder', description: 'Local folder to be synced', required: true })
+  public folder!: string
+
+  @Option(stampProperties)
+  public stamp!: string
+
+  @Option({
+    key: 'remove',
+    type: 'boolean',
+    description: 'Remove paths that do not exist locally',
+  })
+  public remove!: boolean
+
+  public async run(): Promise<void> {
+    await super.init()
+
+    if (!this.stamp) {
+      this.stamp = await pickStamp(this.beeDebug, this.console)
+    }
+    const node = await this.initializeNode(this.reference)
+    const files = await readdirDeepAsync(this.folder, this.folder)
+    const forks = this.getValueForkMap(this.findAllValueForks(node))
+    for (const file of files) {
+      const fork = forks[file]
+
+      if (fork) {
+        fork.found = true
+        const remoteData = await this.bee.downloadData(Buffer.from(fork.node.getEntry).toString('hex'))
+        const localData = readFileSync(this.folder + '/' + file)
+
+        if (localData.equals(remoteData)) {
```

Comment:
it is quite heavy operation. That would be more performant _later when we have the chunker package_ is creating BMT hash of the local file and match it with the corresponding manifest node's entry, thereby you don't have to download the files into memory.

Reply:
so please add a TODO here about that